src/data/real/basic.lean

```diff
@@ -603,6 +603,50 @@ by rw [mul_comm, sqrt_mul' _ hx, mul_comm]
 @[simp] theorem sqrt_div (hx : 0 ≤ x) (y : ℝ) : sqrt (x / y) = sqrt x / sqrt y :=
 by rw [division_def, sqrt_mul hx, sqrt_inv]; refl
 
+/-- If a polynomial of degree 2 is always nonnegative, then its dicriminant is nonpositive -/
+lemma discriminant_le_zero {a b c : ℝ} (h : ∀ x : ℝ,  0 ≤ a*x*x + b*x + c) : b*b - 4*a*c ≤ 0 :=
```

Comment:
Possibly the theorem should be split up. Perhaps, first prove the formula for the completing the square, then prove the formula for the solution of a quadratic (over a field, include a variable `s` for the square root and the assumption `s = b * b - 4 * a * c`), and prove no roots if there's no square root and exactly one root if the discriminant is zero. Then prove when this theorem for strict inequalities, and the hopefully this theorem should be easier to deduce. This is true over any ordered field, and should be proved in that generality.

Reply:
I have generalized the statement to ordered field. It is not too difficult based on the original proof. But for completing the square and the formula for the solution of a quadratic, I hope that I can wait until we have a `field` tactic. Right now I feel it is too much trouble to verify equalities and inequalities on fields. 

I'm not sure where this should go in the library... The file `polynomial` treats polynomials abstractly, so I think this theorem doesn't fit in. The file `ordered_field` is used in the `ring` tactic and the `linarith` tactic, so this cannot go there either. Maybe I need to create a new file?